framework/deproxy_manager.py

```diff
@@ -0,0 +1,69 @@
+import Queue
+import threading
+import asyncore
+import select
+import time
+
+from helpers import stateful, tf_cfg
+
+__author__ = 'Tempesta Technologies, Inc.'
+__copyright__ = 'Copyright (C) 2018 Tempesta Technologies, Inc.'
+__license__ = 'GPL2'
+
+def finish_all_deproxy():
+    asyncore.close_all()
+
+def run_deproxy_server(deproxy, exit_event, is_polling, sockets_changing):
+    tf_cfg.dbg(3, "Running deproxy server manager")
+
+    if hasattr(select, 'poll'):
+        poll_fun = asyncore.poll2
+    else:
+        poll_fun = asyncore.poll
+    while not exit_event.is_set():
+        while sockets_changing.is_set() and not exit_event.is_set():
+            pass
+        if exit_event.is_set():
+            break
+        is_polling.set()
+        poll_fun()
+        is_polling.clear()
+        # servers need some time for locking
+        time.sleep(0.0001)
```

Comment:
Timeout based synchronisation can be buggy, especially with such small timeouts, it's better to wait until some condition is met.